test/e2e/e2e.go

```diff
@@ -0,0 +1,198 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		imageOptions := []string{"-t", c.controllerImageName}
+		err = kubebuilderTest.BuildImage(imageOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating config")
+		configOptions := []string{"--controller-image", c.controllerImageName, "--name", c.installName}
+		err = kubebuilderTest.CreateConfig(configOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("installing controller-manager in cluster")
+		inputFile := filepath.Join(kubebuilderTest.Dir, "hack", "install.yaml")
+		installOptions := []string{"apply", "-f", inputFile}
+		_, err = kubebuilderTest.RunKubectlCommand(framework.GetKubectlArgs(installOptions))
+		Expect(err).NotTo(HaveOccurred())
+
+		By("validate the controller-manager pod running as expected")
+		verifyContollerUp := func() (bool, error) {
+			e = nil
+			// Get pod name
+			getOptions := []string{"get", "pods", "-n", c.namespace, "-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}{{ \"\\n\" }}{{ end }}{{ end }}"}
+			podOutput, err := kubebuilderTest.RunKubectlCommand(framework.GetKubectlArgs(getOptions))
+			Expect(err).NotTo(HaveOccurred())
+			// TODO: validate pod replicas if not default to 1
+			podNames := framework.ParseCmdOutput(podOutput)
+			if len(podNames) != 1 {
+				e = fmt.Errorf("expect 1 controller pods running, but got %d", len(podNames))
+				return false, nil
+			}
+			controllerPodName = podNames[0]
+
+			// Validate pod status
+			getOptions = []string{"get", "pods", controllerPodName, "-n", c.namespace, "-o", "jsonpath={.status.phase}"}
+			status, err := kubebuilderTest.RunKubectlCommand(framework.GetKubectlArgs(getOptions))
+			Expect(err).NotTo(HaveOccurred())
+			if status != "Running" {
+				e = fmt.Errorf("controller pod in %s status", status)
+				return false, nil
+			}
+
+			return true, nil
+		}
+		varifyErr := wait.PollImmediate(500*time.Millisecond, 1*time.Minute, verifyContollerUp)
+		if varifyErr != nil {
+			framework.Failf(e.Error())
+		}
+
+		By("creating resource object")
+		inputFile = filepath.Join(kubebuilderTest.Dir, "hack", "sample", strings.ToLower(c.kind)+".yaml")
+		createOptions := []string{"create", "-f", inputFile}
+		_, err = kubebuilderTest.RunKubectlCommand(framework.GetKubectlArgs(createOptions))
+		Expect(err).NotTo(HaveOccurred())
+
+		By("validate the created resource object gets reconciled in controller")
+		verifyResourceReconciled := func() (bool, error) {
+			// Check container log to validate that the created resource object gets reconciled in controller
+			logOptions := []string{"logs", controllerPodName, "-n", c.namespace}
+			logOutput, err := kubebuilderTest.RunKubectlCommand(framework.GetKubectlArgs(logOptions))
+			Expect(err).NotTo(HaveOccurred())
+
+			if !strings.Contains(logOutput, fmt.Sprintf("to reconcile %s-example", strings.ToLower(c.kind))) {
```

Comment:
I think there is a ginkgo matcher we could use for this - e.g. https://onsi.github.io/gomega/#containsubstringsubstr-string-args-interface